openff/evaluator/utils/openmm.py

```diff
@@ -51,43 +54,54 @@ def setup_platform_with_resources(compute_resources, high_precision=False):
     # Setup the requested platform:
     if compute_resources.number_of_gpus > 0:
 
-        # TODO: Make sure use mixing precision - CUDA, OpenCL.
         # TODO: Deterministic forces = True
 
         from openff.evaluator.backends import ComputeResources
 
         toolkit_enum = ComputeResources.GPUToolkit(
             compute_resources.preferred_gpu_toolkit
         )
+        precision_level = ComputeResources.GPUPrecision(
+            compute_resources.preferred_gpu_precision
+        ).name
 
-        # A platform which runs on GPUs has been requested.
-        platform_name = (
-            "CUDA"
-            if toolkit_enum == ComputeResources.GPUToolkit.CUDA
-            else ComputeResources.GPUToolkit.OpenCL
-        )
+        # Get platform for running on GPUs.
+        if toolkit_enum == ComputeResources.GPUToolkit.auto:
 
-        # noinspection PyCallByClass,PyTypeChecker
-        platform = Platform.getPlatformByName(platform_name)
+            from openmmtools.utils import get_fastest_platform
 
-        if compute_resources.gpu_device_indices is not None:
+            # noinspection PyCallByClass,PyTypeChecker
+            platform = get_fastest_platform(minimum_precision=precision_level)
 
-            property_platform_name = platform_name
+        else:
 
-            if toolkit_enum == ComputeResources.GPUToolkit.CUDA:
-                property_platform_name = platform_name.lower().capitalize()
+            platform_name = "CUDA"
```

Comment:
Defaulting to `CUDA` here? Perhaps clearer to do an `else, else if`?

Reply:
Agreed, I've made this clearer and included an exception if the `toolkit_enum` is not supported.